src/profile/notification-settings/PreferenceCard.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { CSSProperties, ReactElement } from 'react';
+import { EventWorkspaceAttributes } from 'src/libs/events';
+import { memoWithName } from 'src/libs/react-utils';
+import * as Style from 'src/libs/style';
+import {
+  NotificationCheckbox,
+  NotificationCheckboxProps,
+  UserAttributesCheckbox,
+  UserAttributesCheckboxProps,
+} from 'src/profile/notification-settings/PreferenceCheckbox';
+import { NotificationType } from 'src/profile/notification-settings/utils';
+
+const cardStyles: Record<string, CSSProperties> = {
+  label: {
+    ...Style.noWrapEllipsis,
+    height: '1rem',
+    gridColumnStart: '1',
+    gridColumnEnd: '3',
+  },
+  field: {
+    height: '1rem',
+    textAlign: 'center',
+  },
+  row: { display: 'grid', gridTemplateColumns: 'repeat(4, 1fr)', width: '100%', padding: '1rem' },
+};
+
+export interface NotificationCardProps {
+  label: string;
+  setSaving: (boolean) => void;
+  prefsData: Record<string, string>;
+  workspace?: EventWorkspaceAttributes;
+  options: { notificationKeys: string[]; notificationType: NotificationType; optionLabel: string }[];
+}
+
+export const NotificationCard: React.FC<NotificationCardProps> = memoWithName(
```

Comment:
Taken from https://github.com/DataBiosphere/terra-ui/blob/366a30ae3ae2b263355581e63a45de9eebc8017c/src/profile/notification-settings/NotificationSettings.js#L44, modified to be able to take multiple options and converted to TSX.